cvprac/cvp_api.py

```diff
@@ -361,6 +362,86 @@ def get_inventory(self, start=0, end=0, query=''):
                     dev['containerName'] = ''
             return data
 
+    def add_devices_to_inventory(self, device_list):
+        ''' Add a list of devices to the specified parent container.
+
+            Args:
+                device_list (list): A list of devices to be added in the
+                    form of dictionaries. Each device dictionary should
+                    contain the following information:
+                    - device_ip (str): ip address of device we are adding
+                    - parent_name (str): Parent container name
+                    - parent_key (str): Parent container key
+
+                    Example:
+                    device_list = [
+                        {
+                            device_ip: '10.10.10.1',
+                            parent_name: 'Tenant',
+                            parent_key: 'root'
+                        },
+                        {
+                            device_ip: '10.10.10.2',
+                            parent_name: 'MyContainer',
+                            parent_key: 'container-id-1234'
+                        }
+                    ]
+        '''
+
+        self.log.debug('add_device_to_inventory: called')
+        if self.clnt.apiversion is None:
+            self.get_cvp_info()
+        if self.clnt.apiversion == 'v1':
+            self.log.debug('v1 Inventory API Call')
+            data_list = []
+            for device in device_list:
+                dev_data = {
+                    'containerName': device['parent_name'],
+                    'containerId': device['parent_key'],
+                    'containerType': 'Existing',
+                    'ipAddress': device['device_ip'],
+                    'containerList': []
+                }
+                data_list.append(dev_data)
+            data = {'data': data_list}
+            self.clnt.post('/inventory/add/addToInventory.do?'
+                           'startIndex=0&endIndex=0', data=data,
+                           timeout=self.request_timeout)
+        else:
+            self.log.debug('v2 Inventory API Call')
+
+            # First add the devices to the inventory in a single call
+            data = {'hosts': [x['device_ip'] for x in device_list]}
+            self.clnt.post('/inventory/devices', data=data,
+                           timeout=self.request_timeout)
+
+            # With v2, the devices can take a few moments to appear
+            # We need them present before we can move them to a container
+            device_ips = [x['device_ip'] for x in device_list]
+            inv = self.get_inventory()
+            timeout = time.time() + 600
```

Comment:
In general if there is a potential timeout issue for v2 between adding of a device to inventory and moving it to its parent container shouldn't this timeout be added or optional with the single device version of this method? Is this only an issue when multiple devices are added to inventory in a single call?

Reply:
Yes, the timeout should be optional for both single and multiple device versions. Meant to add that in as a 'wait' option, defaulting to False.

Also thinking maybe single device version should just call multiple device version with a one element list. Does that make sense?

Reply:
Ya I think that makes sense and would allow the multiple device function to leverage the current single device function systest.